modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/serializer/RecordSerializerFactoryImpl.java

```diff
@@ -45,22 +46,40 @@
     /** Skip position check flag. Should be set for reading compacted wal file with skipped physical records. */
     private boolean skipPositionCheck;
 
+    /** Key provider for WAL records relating with encrypted caches. */
+    private final EncryptionCacheKeyProvider cacheKeyProvider;
+
     /**
-     * @param cctx Cctx.
+     * @param cctx Cache shared context.
      */
     public RecordSerializerFactoryImpl(GridCacheSharedContext cctx) {
         this(cctx, null);
     }
 
     /**
-     * @param cctx Cctx.
+     * @param cctx Cache shared context.
+     * @param readTypeFilter Type filter.
      */
     public RecordSerializerFactoryImpl(
         GridCacheSharedContext cctx,
         @Nullable IgniteBiPredicate<WALRecord.RecordType, WALPointer> readTypeFilter
+    ) {
+        this(cctx, readTypeFilter, null);
```

Comment:
I would prefer to get a key provider here instead of calling the constructor with `null` value `this(cctx, readTypeFilter, cctx.kernalContext().encryption());`
and add the corresponding assert to the below constructor `assert cacheKeyProvider != null : Encryption key's provider must be initialized.`

Anyway, it is up to you.